packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -2233,7 +2238,10 @@ function commitLayoutEffects(root: FiberRoot, committedLanes: Lanes) {
       commitLayoutEffectOnFiber(root, current, nextEffect, committedLanes);
     }
 
-    if (effectTag & Ref) {
+    if (
+      effectTag & Ref &&
+      (!enableScopeAPI || nextEffect.tag !== ScopeComponent)
+    ) {
```

Comment:
IMO I think this would be a bit more readable if the feature flag check was separated, e.g.
```suggestion
if (enableScopeAPI) {
  if ((effectTag & Ref) !== NoEffect && nextEffect.tag !== ScopeComponent) {
    commitAttachRef(nextEffect);
  }
} else {
  if ((effectTag & Ref) !== NoEffect) {
    commitAttachRef(nextEffect);
  }
}
```